planner/core/rule_column_pruning.go

```diff
@@ -303,24 +303,7 @@ func (p *LogicalJoin) extractUsedCols(parentUsedCols []*expression.Column) (left
 }
 
 func (p *LogicalJoin) mergeSchema() {
-	lChild := p.children[0]
-	rChild := p.children[1]
-	if p.JoinType == SemiJoin || p.JoinType == AntiSemiJoin {
-		p.schema = lChild.Schema().Clone()
-	} else if p.JoinType == LeftOuterSemiJoin || p.JoinType == AntiLeftOuterSemiJoin {
-		joinCol := p.schema.Columns[len(p.schema.Columns)-1]
-		p.schema = lChild.Schema().Clone()
-		p.schema.Append(joinCol)
-	} else {
-		p.schema = expression.MergeSchema(lChild.Schema(), rChild.Schema())
-		switch p.JoinType {
-		case LeftOuterJoin:
-			resetNotNullFlag(p.schema, p.children[1].Schema().Len(), p.schema.Len())
```

Comment:
`p.children[1].Schema().Len()` should be `p.children[0].Schema().Len()`

Reply:
ref https://github.com/pingcap/tidb/blob/505955ad0b6bdaf2550c1bd1ab4858834af02048/planner/core/util.go#L228